c/httpfileservice.c

```diff
@@ -452,6 +460,76 @@ void respondWithUnixFileMetadata(HttpResponse *response, char *absolutePath) {
   }
 }
 
+void directoryChangeOwnerAndRespond(HttpResponse *response, char *path,
+        char *usrId, char *grpId, char *Recursive, char *pattern) {
```

Comment:
Why do we pass strings for group ID, user ID and Recursive? Since this is library code, it would make more sense to pass the expected types, i.e. int and bool. That way this function wouldn't have to do additional data validation and would just change the owner.

Reply:
This is following the man page definitions where either name or numerical values may be passed.  As such, the comparison must be done somewhere.  I chose to do it in the common library where it is closer to low level functions which returns the user and group information.

Reply:
Can you please provide the link?

Here's what I found from IBM, https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.2.0/com.ibm.zos.v2r2.bpxbd00/rtcho.htm

Reply:
https://linuxize.com/post/linux-chown-command/

And in the linux implementation, the uid parameter may be the user id or the name of the user.
So the question is, what standard are we following.
I ask Sean this and he requested that we allow the user name to be passed (also the group).  Note the current implementation allows you to pass in either.

Reply:
On z/OS the definitive answer is sometimes in the low-level UNIX assembler services manual [chown](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.1.0/com.ibm.zos.v2r1.bpxb100/cho.htm)

Reply:
I think more logical would be:
* The ZSS service accepts both user name and UID (i.e. behaves like `chown` program)
* The low level API in zowe-common-c accepts the UID and GID types (i.e. behaves as the `chown` library function, like [UNIX chown](http://man7.org/linux/man-pages/man2/chown.2.html) or  [IBM chown](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.2.0/com.ibm.zos.v2r2.bpxbd00/rtcho.htm) do)

Reply:
@sbollingerAtRocket what do you think about our proposal?

Reply:
I already have a functions that is almost this: 
fileChangeOwner(char *fileName, int *returnCode, int *reasonCode, int usrId, int grpID)
I can easily create function chown function that ignores the return and reason code.

Currently, the zss server accepts a user and group variable as either an integer value or the name of a user/group.  The real question is where the conversion to integer values occurs.
Currently, it is being done in the zowe-common and not in zss.    Even if the conversion was done in the zss, it still would not be calling the low level function directly.  This interface supports recursion and pattern checking and that would still be done in the zowe-common.

Reply:
At this level of the logic, it is the best user experience to support both IDs and names, as zos as well as every other OS I have ever seen that has chown does have a way to specify names. Certainly, there may be a low-level call that only uses IDs, but it's a convenience here that we support both and convert for compatibility when calling down below.